### PR TITLE
updated schema with updateCourse updateStudent and updateUser

### PR DESCRIPTION
**Pull request by ssharm1524**

### CODE REVIEW

File: react-app/src/backend/FirestoreCalls.ts [hunk 1]
```diff
@@ -89,3 +91,61 @@ export function deleteStudent(id: string): Promise<void> {
...
+    firstName: student.firstName,
+    middleName: student?.middleName,
+    lastName: student.lastName,
+    addrFirstLine: student.addrFirstLine,
+    addrSecondLine: student?.addrSecondLine,
+    city: student.city,
+    state: student.state,
+    zipCode: student.zipCode,
+    email: student.email,
+    birthdate: student.birthDate,
```
Comment (sophtsai) [inline react-app/src/backend/FirestoreCalls.ts:111]: "birthdate" capitalization -> should be "birthDate"

File: react-app/src/backend/FirestoreCalls.ts [hunk 2]
```diff
@@ -89,3 +91,61 @@ export function deleteStudent(id: string): Promise<void> {
...
   });
 }
+
+export function updateStudent(student: Student, id: string): Promise<void> {
+  if (!id) {
+    return Promise.reject(new Error('Invalid id'));
+  }
+
+  const studentRef = doc(db, 'Students', id);
+  return updateDoc(studentRef, {
```
Comment (sophtsai) [inline react-app/src/backend/FirestoreCalls.ts:101]: Add a `.then() .catch()` structure to this as well

File: react-app/src/backend/FirestoreCalls.ts [hunk 3]
```diff
@@ -4,11 +4,13 @@ import {
   addDoc,
   deleteDoc,
   getDocs,
+  updateDoc,
 } from 'firebase/firestore';
 import { db } from '../config/firebase';
 import { type Student } from '../types/StudentType';
 import { type Course } from '../types/CourseType';
 import { type Teacher, type TeacherCourse } from '../types/UserType';
```
Comment (sophtsai) [inline react-app/src/backend/FirestoreCalls.ts:12]: You can just append the YKNOTUser type in this line

**Review by sophtsai — CHANGES_REQUESTED** [review]

**Review by sahilg24 — APPROVED** [review]
Looks good!

**Pull request merged**